pubnub-gson/src/main/java/com/pubnub/internal/PubNubImpl.java

```diff
@@ -0,0 +1,607 @@
+        return decryptInputStream(inputStream, null);
+    }
+
+    @Override
+    public @NotNull InputStream decryptInputStream(@NotNull InputStream inputStream, @Nullable String cipherKey) throws PubNubException {
+        return getPubNubCore().decryptInputStream(inputStream, cipherKey);
+    }
+
+    /**
+     * Perform Cryptographic encryption of an input string and the cipher key provided by PNConfiguration
+     *
+     * @param inputString String to be encrypted
+     * @return String containing the encryption of inputString using cipherKey
+     */
+    @Override
+    public @NotNull String encrypt(@NotNull String inputString) throws PubNubException {
+        return encrypt(inputString, null);
+    }
+
+    /**
+     * Perform Cryptographic encryption of an input string and the cipher key.
+     *
+     * @param inputString String to be encrypted
+     * @param cipherKey   cipher key to be used for encryption
+     * @return String containing the encryption of inputString using cipherKey
+     * @throws PubNubException throws exception in case of failed encryption
+     */
+    @Override
+    public @NotNull String encrypt(@NotNull String inputString, String cipherKey) throws PubNubException {
+        return getPubNubCore().encrypt(inputString, cipherKey);
+    }
+
+    @Override
+    public @NotNull InputStream encryptInputStream(@NotNull InputStream inputStream) throws PubNubException {
+        return encryptInputStream(inputStream, null);
+    }
+
+    @Override
+    public @NotNull InputStream encryptInputStream(@NotNull InputStream inputStream, String cipherKey) throws PubNubException {
+        return getPubNubCore().encryptInputStream(inputStream, cipherKey);
+    }
+
+    @Override
+    @NotNull
+    public Publish fire() {
+        return publish().shouldStore(false).replicate(false);
+    }
+
+    @Override
+    @NotNull
+    public List<String> getSubscribedChannels() {
+        return getPubNubCore().getSubscribedChannels();
+    }
+
+    @Override
+    @NotNull
+    public List<String> getSubscribedChannelGroups() {
+        return getPubNubCore().getSubscribedChannelGroups();
+    }
+
+    @NotNull
+    @Override
+    public Channel channel(@NotNull String name) {
+        return new ChannelImpl(this, name);
+    }
+
+    @NotNull
+    @Override
+    public ChannelGroup channelGroup(@NotNull String name) {
+        return new ChannelGroupImpl(this, name);
+    }
+
+    @NotNull
+    @Override
+    public ChannelMetadata channelMetadata(@NotNull String id) {
+        return new ChannelMetadataImpl(this, id);
+    }
+
+    @NotNull
+    @Override
+    public UserMetadata userMetadata(@NotNull String id) {
+        return new UserMetadataImpl(this, id);
+    }
+
+    @NotNull
+    @Override
+    public SubscriptionSet subscriptionSetOf(@NotNull Set<? extends Subscription> subscriptions) {
+        return new SubscriptionSetImpl(getPubNubCore(), Collections.emptySet());
```

Comment:
Why there is:
`Collections.emptySet() `
? 
I guess current shape method parameter "subscriptions" is not needed.
There is SubscriptionImpl#plus that uses `subscriptionSetOf` method